ruby/ql/lib/codeql/ruby/regexp/internal/RegExpTracking.qll

```diff
@@ -17,7 +17,7 @@ private import codeql.ruby.AST as Ast
 private import codeql.ruby.CFG
 private import codeql.ruby.DataFlow
 private import codeql.ruby.controlflow.CfgNodes
-private import codeql.ruby.typetracking.TypeTracker
+private import codeql.ruby.typetracking.internal.TypeTrackingImpl
```

Comment:
Does this really need to import the `Impl` library? I think ideally the Impl module is only used when instantiating the library. The modules that are built on top should ideally only use the "public facing" bits.

Reply:
It does, because the code refers to `StepSummary`, which is considered internal in the new library.

Reply:
Ok, I guess we cannot avoid that, can we? Or should StepSummary perhaps be part of the public interface?

Reply:
I prefer to keep it internal, at least for now. It is not something I expect we need to use very often.